src/main/java/com/codegym/shopyy/controller/api/ProductController.java

```diff
@@ -0,0 +1,146 @@
+package com.codegym.shopyy.controller.api;
+
+import com.codegym.shopyy.dto.request.ProductRequestDto;
+import com.codegym.shopyy.dto.response.ResponsePage;
+import com.codegym.shopyy.model.Product;
+import com.codegym.shopyy.service.impl.ProductServiceImpl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/products")
+@CrossOrigin(origins = "*")
+public class ProductController {
+
+    private static final int DEFAULT_PAGE = 0;

```

Comment:
create package constant, create class PageConstant, move DEFAULT_PAGE, PAGE_SIZE from ProductController class to PageConstant class

Reply:
complete